### DISCUSSION TITLE
[bug]: visual issue

### DISCUSSION THREAD

**Discussion by asumbek** [z-shell/zi#279] [Q&A] [answered]
### Environment

Arch Linux proot inside Termux, zsh 5.9 (armv7l-unknown-linux-gnueabihf), zi git commit 75929e8

### Reproduction steps

```bash
`zi -h` or `exec zsh` to install plugins/snippets
```

### Expected behavior

```bash
1. No error is shown while showing command usage.
2. Git clone progress bar not printed every line.
```

### Current behavior

```bash
1. Errors about `character not in range` printed when executing zi with `-h` flag. (See screenshot)
2. The Git clone progress bar printed every 
line. (See screenshot)
```

### Code snippet

```zsh
{# OMZ plugins
}

{# plugins
    zi load "agkozak/agkozak-zsh-prompt"; {
        bindkey -v

        AGKOZAK_CMD_EXEC_TIME=3

        AGKOZAK_COLORS_CMD_EXEC_TIME=117
        AGKOZAK_COLORS_EXIT_STATUS=196
        AGKOZAK_COLORS_PATH=69
        AGKOZAK_COLORS_BRANCH_STATUS=76
        AGKOZAKTIDE_COLORS_PROMPT_NORMAL=76
        AGKOZAKTIDE_COLORS_PROMPT_VI=45
        AGKOZAKTIDE_COLORS_PROMPT_NONZERO=196
        AGKOZAKTIDE_COLORS_CLOCK=default

        AGKOZAK_CUSTOM_PROMPT=$'\n'
        # Directory
        AGKOZAK_CUSTOM_PROMPT+=$'%B%F{$AGKOZAK_COLORS_PATH}%2v%f%b'
        # Exec time
        AGKOZAK_CUSTOM_PROMPT+=$'%(9V.\n%F{$AGKOZAK_COLORS_CMD_EXEC_TIME}%9v :: %f.)'
        # Exit code & Job
        AGKOZAK_CUSTOM_PROMPT+=$'%(?..%(9V..\n)[%F{$AGKOZAK_COLORS_EXIT_STATUS}:%?%f%(1j..]))%(1j.%(?.\n[. )%F{$AGKOZAK_COLORS_BG_STRING}*%j%f].)'
        # Prompt
        AGKOZAK_CUSTOM_PROMPT+=$'\n%(?.%(4V.%F{$AGKOZAKTIDE_COLORS_PROMPT_VI}≈>%f.%F{$AGKOZAKTIDE_COLORS_PROMPT_NORMAL}=>%f).%F{$AGKOZAKTIDE_COLORS_PROMPT_NONZERO}=>%f) '

        AGKOZAK_CUSTOM_RPROMPT=""
        # Git status
        AGKOZAK_CUSTOM_RPROMPT+=$'%(3V.%F{${AGKOZAK_COLORS_BRANCH_STATUS}}%6v%(7V. %7v.)%f.) '
        # Clock
        AGKOZAK_CUSTOM_RPROMPT+="%F{$AGKOZAKTIDE_COLORS_CLOCK}%*%f"
    }

    zi ice pick"themes/catppuccin_mocha-zsh-syntax-highlighting.zsh"
    zi load "catppuccin/zsh-syntax-highlighting"

    zi ice pick"zsh-syntax-highlighting.zsh"
    zi load "zsh-users/zsh-syntax-highlighting"

    zi ice pick"zsh-autosuggestions.zsh"
    zi load "zsh-users/zsh-autosuggestions"; {
        ZSH_AUTOSUGGEST_HIGHLIGHT_STYLE="fg=#9399b2"
    }

    zi ice pick"zsh-history-substring-search.zsh"
    zi load "zsh-users/zsh-history-substring-search"; {
        bindkey '^[[A' history-substring-search-up
        bindkey '^[[B' history-substring-search-down

        HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_FOUND="underline,bold"
        HISTORY_SUBSTRING_SEARCH_HIGHLIGHT_NOT_FOUND="fg=1,underline,bold"
    }

    zi load "skywind3000/z.lua"; {}

    zi load "davidde/git"; {}

    zi load "cjayross/up"; {}

    zi load "jeffreytse/zsh-vi-mode"; {}

    zi load "hcgraf/zsh-sudo"; {}
}
```

### Additional information

Screenshots:
![Screenshot_20230711-170945_Termux](https://github.com/z-shell/zi/assets/71205197/821ae55a-f50d-43dc-9912-4ce45b3074c5)
![Screenshot_20230711-170617_Termux](https://github.com/z-shell/zi/assets/71205197/41da8e89-f87b-4603-90fc-ef10ef358aad)

### Self-service

- [ ] I'd be willing to address this documentation request myself.

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/z-shell/community/blob/main/docs/CONTRIBUTING_GUIDELINES.md).

### Are you familiar with the Contributor Covenant Code of Conduct?

- [X] I have read the [Contributor Covenant Code of Conduct](https://github.com/z-shell/zi/blob/main/docs/CODE_OF_CONDUCT.md).

```[tasklist]
### Tasks
```

**Comment by ss-o** [comment]
Hi @UrNightmaree,

Thank for opening an issue :handshake: I have scheduled a triage and will update you on the progress :dart: 

Please provide information about your terminal emulator :grey_question: 

Thanks.

**Comment by asumbek** [comment]
Hello @ss-o, thank you for responding my issue. 🙏

> Please provide information about your terminal emulator ❔

It's Termux.

**Comment by ss-o** [comment]
It seems to be a locale issue and encounters an [escape sequence](https://github.com/z-shell/zi/blob/75929e8f4425761a64591d0f6c23dbb0c64261fc/zi.zsh#L1768C44-L1768C44) which most likely is not valid for currently set encoding. To verify this you may try:

```shell
LC_ALL="UTF-8" echo '\u2603'
LC_CTYPE="UTF-8" echo '\u2603'
LC_CTYPE="en_US.UTF-8" echo "\u2603"
```

The above should print a symbol or if due to missing fonts may provide empty/incorrect output, but no error should be shown, but with the following example it should show error "character not in range":

```shell
LC_CTYPE="C" echo '\u2603'
```

Could you please confirm that this is the case? 

> Source related to locale: https://wiki.archlinux.org/title/locale

**Comment by asumbek** [comment]
I tried above but still complain about `character not in range`.
![Screenshot_20230810-101858_Termux](https://github.com/z-shell/zi/assets/71205197/ba67324e-1540-46e6-8f2f-4db99575835b)

**Comment by ss-o** [comment, marked as answer]
Pretty sure that the error is shown because of the locale and multiple scenarios may cause this, e.g. lack of configuration, misconfiguration, corrupted files, etc.

Try the following:

Open the `/etc/locale.gen` file and ensure there is a line as below and it is not prepended with a `#`, then run `locale-gen` (usually requires root access and executed with `sudo`)

```ini
en_US.UTF-8 UTF-8
```

Add at the top of your `.zshrc`, then run `exec zsh`.

```shell
export LANG=en_US.UTF-8
export LC_ALL=en_US.UTF-8
```

---

If the issue is not solved then check if it is caused by something in `.zshrc`:

```shell
mv ~/.zshrc ~/.zshrc.backup
touch ~/.zshrc
exec zsh
echo '\u2603'
```

If the `character not in range` error does not appear then recover `.zshrc` with `cp -i ~/.zshrc.backup ~/.zshrc` and remove components one by one while each time running `exec zsh` and `echo '\u2603'` afterward to identify what exactly causes the error.

**Comment by asumbek** [comment]
Oh now it works, seems like I haven't setup locale.
